From c2f8c8ba3f07bb545d13aa09ccb46a1266032fd6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alex=20R=C3=B8nne=20Petersen?= <alex@alexrp.com>
Date: Thu, 4 Sep 2025 00:50:34 +0200
Subject: [PATCH 091/133] std.elf: change STV enum from u2 to u3
X-Developer-Signature: v=1; a=openpgp-sha256; l=2085; i=xtex@astrafall.org;
 h=from:subject; bh=h7bJGCMeNdomzCFSMGxxElRH4e90VcB4cHHkEaPLOfQ=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78jhuPzv1m/kly2rDE38q7dNFt265dFvilfd59lBPj
 svhh4vKOiayMIhxMViKKbIUGTZ4s+qk84suK5eFmcPKBDJEWqSBgYGBgYWBLzcxr9RIx0jPVNtQ
 z9BQx0jHiIGLUwCmOj2RkeFS0eUvq2d4vz525OG6bWFPTSzfM2R9cLE4eM32xrHpWe6BDL9ZmHZ
 PXz1N8+9Lm+BL5Yc3rZ024RJPxo1pF9W2/+jYZMjLAgA=
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

In gABI 4.3, st_other was changed such that the lower 3 bits are reserved for
the visibility, up from the previous 2 bits.

(cherry picked from commit a82f446d91c9e764719d34188de5d794ee24dda8)
---
 lib/std/Build/Step/CheckObject.zig | 2 +-
 lib/std/elf.zig                    | 2 +-
 tools/gen_stubs.zig                | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/std/Build/Step/CheckObject.zig b/lib/std/Build/Step/CheckObject.zig
index 6bbe3307a0f6..e56a92421181 100644
--- a/lib/std/Build/Step/CheckObject.zig
+++ b/lib/std/Build/Step/CheckObject.zig
@@ -2259,7 +2259,7 @@ const ElfDumper = struct {
                     try writer.print(" {s}", .{sym_bind});
                 }
 
-                const sym_vis = @as(elf.STV, @enumFromInt(@as(u2, @truncate(sym.st_other))));
+                const sym_vis = @as(elf.STV, @enumFromInt(@as(u3, @truncate(sym.st_other))));
                 try writer.print(" {s}", .{@tagName(sym_vis)});
 
                 const sym_name = switch (sym.st_type()) {
diff --git a/lib/std/elf.zig b/lib/std/elf.zig
index b0e5dd8fec5c..6696fbda13e0 100644
--- a/lib/std/elf.zig
+++ b/lib/std/elf.zig
@@ -2303,7 +2303,7 @@ pub const R_PPC64 = enum(u32) {
     _,
 };
 
-pub const STV = enum(u2) {
+pub const STV = enum(u3) {
     DEFAULT = 0,
     INTERNAL = 1,
     HIDDEN = 2,
diff --git a/tools/gen_stubs.zig b/tools/gen_stubs.zig
index 9fdd63eda11e..9f5219d40f15 100644
--- a/tools/gen_stubs.zig
+++ b/tools/gen_stubs.zig
@@ -609,7 +609,7 @@ fn parseElf(parse: Parse, comptime is_64: bool, comptime endian: builtin.Endian)
         const name = try arena.dupe(u8, mem.sliceTo(dynstr[s(sym.st_name)..], 0));
         const ty = @as(u4, @truncate(sym.st_info));
         const binding = @as(u4, @truncate(sym.st_info >> 4));
-        const visib = @as(elf.STV, @enumFromInt(@as(u2, @truncate(sym.st_other))));
+        const visib = @as(elf.STV, @enumFromInt(@as(u3, @truncate(sym.st_other))));
         const size = s(sym.st_size);
 
         if (size == 0) {
-- 
2.52.0

